packages/pyright-internal/src/analyzer/sourceFile.ts

```diff
@@ -520,6 +524,15 @@ export class SourceFile {
             let fileContents = this.getFileContents();
             if (fileContents === undefined) {
                 try {
+                    const fileStat = this.fileSystem.statSync(this._filePath);
```

Comment:
Can you move this section down into the `timeOperation` below? That way, we'll count the time it takes to stat in addition to the read itself and not have unaccounted time.